src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -614,12 +620,12 @@ Status LDAPManagerImpl::execQuery(std::string& ldapurl, std::vector<std::string>
         } else {
             BerElement *ber = nullptr;
             auto attribute = ldap_first_attribute(ldap, entry, &ber);
-            ON_BLOCK_EXIT([&] { ber_free(ber, 0); });
+            ON_BLOCK_EXIT([=] { ber_free(ber, 0); });
             while (attribute) {
-                ON_BLOCK_EXIT([&] { ldap_memfree(attribute); });
+                ON_BLOCK_EXIT([=] { ldap_memfree(attribute); });
 
                 auto const values = ldap_get_values_len(ldap, entry, attribute);
-                ON_BLOCK_EXIT([&] { ldap_value_free_len(values); });
+                ON_BLOCK_EXIT([=] { ldap_value_free_len(values); });
```

Comment:
Would it yield more reliable and easy-to-read source code, if `unique_ptr`s with custom deleters were used instead of the combination of raw pointers and the `ON_BLOCK_EXIT`macros?

Reply:
If there is a desire to replace `ON_BLOCK_EXIT` with `unique_ptr`, it is better to implement that as a dedicated PR.